pkg/kubelet/kubelet_volumes.go

```diff
@@ -58,6 +58,16 @@ func (kl *Kubelet) podVolumesExist(podUID types.UID) bool {
 		return true
 	}
 
+	volumePaths, err := kl.getPodVolumePathListFromDisk(podUID)
+	if err != nil {
+		glog.Errorf("pod %q found, but error %v occurred during reading volume dir from disk", podUID, err)
+		return true
+	}
+	if len(volumePaths) > 0 {
+		glog.Errorf("pod %q found, but volume paths are still present on disk", podUID)
+		return true
+	}
```

Comment:
While this check mostly looks okay, I am wondering if this will prevent deletion of pods that somehow have leftover directories on the node. To be clear - stopping pod deletion for mounted volumes is fine, but I have seen several cases where pods just leave unmounted pod directory structure in-place because of node restart or crash. Will we prevent users from deleting such pods?

cc @jsafrane

Reply:
In that case, since desired state still has the pod at first, reconciler will try to mount the volume and update the actual state. After the volume exists in actual state, since pod is in terminated state, desired_state_populator will delete the pod from the desired state. Then reconciler should try to tear down the volume which remove all the directories. 
Unless something really bad happened, and reconciler fail to mount volume all the time, pod will remains in the desired state. In case, user should use --force to delete the pod.

Reply:
After checking the code, I think it is better to check whether volumes are still mounted instead of just volume paths exist here. So only if volumes are still mounted on the disk, pod should not be deleted. If all mounts are gone, pod could be deleted. The downside of this is that empty directories migh not be cleaned up.